File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> m;
        std::vector<int> v;

        int i = 0;
        for (int num : nums) {
            // Check 1. (m.contains(target - num)) -> error
            // Check 2. (m.find(target - num) != m.end()) -> 26ms
            // Check 3. (m.count(target - num)) -> 16ms
            if (m.find(target - num) != m.end()) {
                return {m[target - num], i};
            } else {
                m.insert(pair(num, i));
            }
            i++;
        }
        return v;
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        int num;
        for (int i = 0; i < nums.length; i++) {
            num = nums[i];
            if (map.containsKey(target - num)) {
                return new int[]{map.get(target - num), i};
            } else {
                map.put(num, i);
            }
        }
        return new int[]{};
    }
}